test/processImage.js

```diff
@@ -953,7 +955,9 @@ describe('express-processimage', () => {
               __dirname,
               '..',
               'testdata',
-              'turtleCroppedCenterGm.jpg'
+              isDarwin
+                ? 'turtleCroppedCenterGm-darwin.jpg'
+                : 'turtleCroppedCenterGm.jpg'
```

Comment:
I guess our alternative is to accept some degree of mismatch. On my mac they come out 13.83% different because of differently colored pixels. The point of the test is to make sure that the image gets cropped correctly, so this would also do the trick:

```js
    it('should allow a crop operation with a gravity of center', () =>
      expect(
        'GET /turtle.jpg?gm&resize=40,15&crop=center',
        'to yield response',
        {
          body: expect.it(
            'to resemble',
            pathModule.resolve(
              __dirname,
              '..',
              'testdata',
              'turtleCroppedCenterGm.jpg'
            ),
            {
              mismatchPercentage: expect.it('to be less than', 15),
            }
          ),
        }
      ));
```

Reply:
Yeah - it’s always the same difference with macOS and this was basically applying the same trick used by impro. I think there’s still value in having the tests stay precise (would it still catch the change in output from the sharp upgrade) and could probably add macOS to the build matrix if you’d prefer it was exercised. Let me know :)

Reply:
I guess it doesn't matter so much, it's just a bit of a sore thumb to have a per platform expected output. It's probably really because of different versions of graphicsmagick or one of the underlying libraries, and what if we suddenly have a 3rd and 4th combination that comes out differently. It also means that it'll be hard to actually maintain the test suite without owning both a linux box and a mac.

It would have been nice if the image diffing was a bit smarter. I looked at the two pictures side by side, and it's only really a slight nuance change to some of the pixels. The mismatch percentage only ends up at 13% because resemble counts the number of differently colored pixels without looking at "how different" the color values are.

I played around with resemble.js a bit just now, and it has the ability to ignore antialiasing, which seems to do the trick. At least it judges `turtleCroppedNorthEastGm.jpg` to be identical to `testdata/turtleCroppedNorthEastGm-darwin.jpg` on my machine when it's on.

So this ought to work:

```js
    it('should allow a crop operation with a gravity of center', () =>
      expect(
        'GET /turtle.jpg?gm&resize=40,15&crop=center',
        'to yield response',
        {
          body: expect.it(
            'to resemble',
            pathModule.resolve(
              __dirname,
              '..',
              'testdata',
              'turtleCroppedCenterGm.jpg'
            ),
            {
              ignoreAntialiasing: true, // Ignore small pixel nuance differences observed on OSX vs. linux
            }
          ),
        }
      ));
```

It didn't work before because of a bug in unexpected-resemble, please upgrade to the (in range) version 4.3.1 that I just published before trying the above :)

> and could probably add macOS to the build matrix if you’d prefer it was exercised

That would be a nice addition no matter what :star_struck: